pkg/jobs/broker.go

```diff
@@ -304,6 +313,66 @@ func GetQueuedJobs(db prefixer.Prefixer, workerType string) ([]*Job, error) {
 	return results, nil
 }
 
+// GetJobsBeforeDate returns alls jobs queued before the specified date
+func GetJobsBeforeDate(db prefixer.Prefixer, date time.Time) ([]*Job, error) {
+	var res couchdb.ViewResponse
+	var jobs []*Job
+
+	err := couchdb.ExecView(db, consts.JobByQueuedAt, &couchdb.ViewRequest{
+		IncludeDocs: true,
+		EndKey:      date.Format(time.RFC3339Nano),
+	}, &res)
+	if err != nil {
+		return nil, err
+	}
+	if len(res.Rows) == 0 {
+		return nil, ErrNotFoundJob
+	}
+
+	// Returning docs
+	doc := &Job{}
```

Comment:
I'd prefer to put this declaration inside the loop. IIRC, `json.Unmarshal` does not reset the document, and we may keep one field from the previous iteration, like `manual_execution` which has `omitempty`.